default.nix

```diff
@@ -320,16 +320,21 @@ in rec {
   };
 
   agdaPackages = rec {
-    # We can use Agda from nixpkgs for the moment, we may need to change this again
-    # if we want to move to a more recent version.
-    Agda = pkgs.haskellPackages.Agda;
-    agda = pkgs.agda;
+    Agda = haskell.extraPackages.Agda.components.exes.agda;
+    agda = pkgs.agda.override {
+      inherit Agda;
+      extension = self: super: {
+        # New Agda needs this to put interface files next to source files. Leaving it like this for now to avoid
+        # writing a new Agda builder - maybe nixpkgs will figure this out for us.
+        buildFlags = super.buildFlags ++ [ "--local-interfaces"];
+      };
+    };
 
     # We also rely on a newer version of the stdlib
-    AgdaStdlib = pkgs.AgdaStdlib.overrideAttrs (oldAttrs: rec {
+    AgdaStdlib = (pkgs.AgdaStdlib.override { inherit agda; }).overrideAttrs (oldAttrs: rec {
       # Need to override the source this way
       name = "agda-stdlib-${version}";
-      version = "1.2";
+      version = "1.3";
       src = sources.agda-stdlib;
       # Marked as broken on darwin in our nixpkgs, but not actually broken,
```

Comment:
Perhaps worth pointing out that this is already merged and likely to appear in the next version of `nixos` if that is indeed the case.